### PR TITLE
Vietnam: optimizations, observed holidays calculation

### PR DESCRIPTION
**Pull request by KJhellico**

### CODE REVIEW

File: holidays/countries/vietnam.py [hunk 1]
```diff
@@ -41,53 +42,40 @@ def _populate(self, year):
...
-            king_hung_date = date(dt.year, dt.month, dt.day)
-            self[king_hung_date] = name
-        else:
-            pass
+            king_hung_date = self.get_solar_date(year, 3, 10)
+            self[king_hung_date] = "Hung Kings Commemoration Day"
 
         # Liberation Day/Reunification Day
-        name = "Liberation Day/Reunification Day"
         libration_date = date(year, APR, 30)
```
Comment (arkid15r) [inline holidays/countries/vietnam.py:69]: As the date objects being used once only I'd  optimize them with `self[date(year, APR, 30)] = "Liberation Day/Reunification Day"` approach.

**Review by arkid15r — COMMENTED** [review]
👍

File: holidays/countries/vietnam.py (hunk 1 above)
Comment (KJhellico) [reply holidays/countries/vietnam.py]: Done. Thank you!

**Comment by dr-prodigy** [comment]
👍

**Pull request merged**